Also, I tried building your branch on OSX and I got this:

```
❯ nix build .#pkgs.bazel_7
error: builder for '/nix/store/203zp1rhskwf6nfgpg4l4s3dh9hlbj5k-bazelDeps.drv' failed with exit code 2;
       last 25 log lines:
       > ERROR: An error occurred during the fetch of repository 'bazel_tools~cc_configure_extension~local_config_cc':
       >    Traceback (most recent call last):
       >        File "/private/var/tmp/_bazel__nixbld2/1f90a375f58569e8f4bcdc6145ad9689/external/bazel_tools/tools/cpp/cc_configure.bzl", line 91, column 33, in cc_autoconf_impl
       >              configure_unix_toolchain(repository_ctx, cpu_value, overriden_tools)
       >   File "/private/var/tmp/_bazel__nixbld2/1f90a375f58569e8f4bcdc6145ad9689/external/bazel_tools/tools/cpp/unix_cc_configure.bzl", line 387, column 46, in configure_unix_toolchain
       >                overriden_tools["ar"] = _find_generic(repository_ctx, "libtool", "LIBTOOL", overriden_tools)
       >   File "/private/var/tmp/_bazel__nixbld2/1f90a375f58569e8f4bcdc6145ad9689/external/bazel_tools/tools/cpp/unix_cc_configure.bzl", line 310, column 32, in _find_generic
       >           auto_configure_fail(msg)
       >       File "/private/var/tmp/_bazel__nixbld2/1f90a375f58569e8f4bcdc6145ad9689/external/bazel_tools/tools/cpp/lib_cc_configure.bzl", line 112, column 9, in auto_configure_fail
       >               fail("\n%sAuto-Configuration Error:%s %s\n" % (red, no_color, msg))
       > Error in fail:
       > Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
       > ERROR: no such package '@@bazel_tools~cc_configure_extension~local_config_cc//':
       > Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
       > ERROR: /private/var/tmp/_bazel__nixbld2/1f90a375f58569e8f4bcdc6145ad9689/external/rules_pkg~/pkg/private/tar/BUILD:45:10: @@rules_pkg~//pkg/private/tar:build_tar depends on @@bazel_tools~cc_configure_extension~local_config_cc//:cc-compiler-darwin_arm64 in repository @@bazel_tools~cc_configure_extension~local_config_cc which failed to fetch. no such package '@@bazel_tools~cc_configure_extension~local_config_cc//':
       > Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
       > ERROR: Analysis of target '//src:bazel_nojdk' failed; build aborted: Analysis failed
       > INFO: Elapsed time: 7.130s
       > INFO: 0 processes.
       > ERROR: Build did NOT complete successfully
       > Loading: 325 packages loaded
       > ERROR: Fetching some target dependencies failed with errors: Analysis of target '//src:bazel_nojdk' failed; build aborted: no such package '@@bazel_tools~cc_configure_extension~local_config_cc//':
       > Auto-Configuration Error: Cannot find libtool or LIBTOOL; either correct your path or set the LIBTOOL environment variable
       > Loading: 325 packages loaded
       >
       For full logs, run 'nix log /nix/store/203zp1rhskwf6nfgpg4l4s3dh9hlbj5k-bazelDeps.drv'.
error: 1 dependencies of derivation '/nix/store/62zfamx14j7077q22arbyywwc3s6cnch-bazel-7.3.1.drv' failed to build
```

After making this change:

```
diff --git a/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix b/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix
index 703f25e34..27746585a 100644
--- a/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix
@@ -35,6 +35,7 @@
   # Apple dependencies
   cctools,
   libcxx,
+  libtool,
   sigtool,
   CoreFoundation,
   CoreServices,
@@ -188,6 +189,7 @@ let
         unzip
         runJdk
         bazelForDeps
+        libtool
       ];
       configurePhase = ''
         runHook preConfigure
```

It built on OSX for me. Testing on an M1 Mac

--- a/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/bazel_7/default.nix
@@ -35,6 +35,7 @@
   # Apple dependencies
   cctools,
   libcxx,
+  libtool,
   sigtool,
   CoreFoundation,
   CoreServices,
@@ -188,7 +189,7 @@ let
         unzip
         runJdk
         bazelForDeps
-      ];
+      ] ++ lib.optional (stdenv.hostPlatform.isDarwin) libtool;
       configurePhase = ''
         runHook preConfigure
 

